packages/core/src/components/Form.tsx

```diff
@@ -280,22 +280,60 @@ export default class Form<
     this.formElement = createRef();
   }
 
-  /** React lifecycle method that gets called before new props are provided, updates the state based on new props. It
-   * will also call the`onChange` handler if the `formData` is modified to add missing default values as part of the
-   * state construction.
+  /**
+   * `getSnapshotBeforeUpdate` is a React lifecycle method that is invoked right before the most recently rendered output is committed to the DOM.
+   * It enables your component to capture current values (e.g., scroll position) before they are potentially changed.
+   *
+   * In this case, it checks if the `formData` prop has changed since the last render. If it has, it computes the next state of the component
+   * using `getStateFromProps` method and returns it along with a `shouldUpdate` flag set to `true`. This ensures that we have the most up-to-date
+   * state ready to be applied in `componentDidUpdate`.
+   *
+   * If `formData` hasn't changed, it simply returns an object with `shouldUpdate` set to `false`, indicating that a state update is not necessary.
    *
-   * @param nextProps - The new set of props about to be applied to the `Form`
+   * @param prevProps - The previous set of props before the update.
+   * @returns Either an object containing the next state and a flag indicating that an update should occur, or an object with a flag indicating that an update is not necessary.
    */
-  UNSAFE_componentWillReceiveProps(nextProps: FormProps<T, S, F>) {
-    const nextState = this.getStateFromProps(nextProps, nextProps.formData);
-    if (
-      !deepEquals(nextState.formData, nextProps.formData) &&
-      !deepEquals(nextState.formData, this.state.formData) &&
-      nextProps.onChange
-    ) {
-      nextProps.onChange(nextState);
+
+  getSnapshotBeforeUpdate(
```

Comment:
Please remove the blank line between the documentation and the function
```suggestion
  getSnapshotBeforeUpdate(
```

Reply:
done